src/display/annotation_layer.js

```diff
@@ -2473,9 +2474,10 @@ class AnnotationLayer {
    * @memberof AnnotationLayer
    */
   static render(parameters) {
-    const { annotations, div, viewport } = parameters;
+    const { annotations, div, viewport, accessibilityManager } = parameters;
 
     this.#setDimensions(div, viewport);
+    let zIndex = 0;
```

Comment:
Can you please elaborate what, and why, all these z-index changes are for?

Reply:
Without this patch annotations are rendered in the order they're in the pdf.
With this patch they're reordered when they're inserted in order to respect the visual order, hence an annotation which should be above an other one, could now be underneath.
The z-index will help to respect the order of the annotation rendering.

Reply:
Thanks for the explanation; some sort of (short) comment about that might not hurt :-)

Reply:
Yep I added one.